protected/humhub/modules/content/widgets/stream/StreamEntryWidget.php

```diff
@@ -106,13 +106,19 @@ private static function renderLegacyWallEntry(ContentActiveRecord $record, $opti
             $jsWidget = $options['jsWidget'];
             unset($options['jsWidget']);
         } else {
-            $jsWidget = $record->getWallEntryWidget()->jsWidget;
+            $wallEntryWidget = $record->getWallEntryWidget();
+            $jsWidget = $wallEntryWidget instanceof StreamEntryWidget ? $wallEntryWidget->jsWidget : null;
```

Comment:
Should we really allow non `StreamEntryWidget`  in the stream?

Reply:
Same as https://github.com/humhub/humhub/pull/6941#discussion_r1566484418

Reply:
@luke- If we should not allow such records in a search stream then we should exclude them from indexing [here](https://github.com/humhub/humhub/blob/develop/protected/humhub/modules/content/services/ContentSearchService.php#L80), otherwise we may have such issues from external modules like the current one.

Reply:
> @luke- If we should not allow such records in a search stream then we should exclude them from indexing [here](https://github.com/humhub/humhub/blob/develop/protected/humhub/modules/content/services/ContentSearchService.php#L80), otherwise we may have such issues from external modules like the current one.

I agree here, maybe we could include a `$exclude = [];` for such cases?

Reply:
@yurabakhtin Can't we just additionally check here whether the Content AR provides a Wall Entry class?

If this is null, we can skip the indexing.  

https://github.com/humhub/humhub/blob/develop/protected/humhub/modules/content/services/ContentSearchService.php#L80

Reply:
@luke- Yes, it is possible to check there by code like `empty($this->content->getModel()->wallEntryClass)`, but this code will does an additional database query to get `ContentActiveRecord`, it may does the indexing process slower.

I think better solution is to fix all wrong ContentActiveRecords which have `$streamChannel = 'default'` and have no `$wallEntryClass`.

Reply:
@luke- Then you can merge this PR, it will informs about such wrong ContentActiveRecords with new log error `Yii::error('Model ' . get_class($record) . ' must define $wallEntryClass or set $streamChannel to null!', 'content');`.